src/System.Net.Http.WinHttpHandler/tests/FunctionalTests/WinHttpHandlerTest.cs

```diff
@@ -39,5 +42,37 @@ public void SendAsync_SimpleGet_Success()
                 _output.WriteLine(responseContent);
             }
         }
+
+        [Fact]
+        [OuterLoop]
+        public void SendAsync_SlowServerAndCancel_ThrowsTaskCanceledException()
+        {
+            var handler = new WinHttpHandler();
+            using (var client = new HttpClient(handler))
+            {
+                var cts = new CancellationTokenSource();
+                Task<HttpResponseMessage> t = client.GetAsync(SlowServer, cts.Token);
+
+                Thread.Sleep(500);
+                cts.Cancel();
+                
+                AggregateException ag = Assert.Throws<AggregateException>(() => t.Wait());
+                Assert.IsType<TaskCanceledException>(ag.InnerException);
+            }
+        }        
```

Comment:
Can we move some of these tests that don't do anything specific to the handler to instead be HttpClient tests?  That'll help to cover both the Windows and Unix implementations and help ensure similar behavior across them.